File: kotlinames-common/src/main/java/com/nagopy/android/kotlinames/property/required/KRequiredShortPropertyName.java
package com.nagopy.android.kotlinames.property.required;

import com.nagopy.android.kotlinames.KRequiredPropertyName;
import com.nagopy.android.kotlinames.KSortablePropertyName;

public class KRequiredShortPropertyName implements KRequiredPropertyName<Short>, KSortablePropertyName<Short> {

    private final String name;

    public KRequiredShortPropertyName(String name) {
        this.name = name;
    }

    @Override
    public String name() {
        return name;
    }
}

File: kotlinames/src/main/kotlin/com/nagopy/android/kotlinames/property/required/KRequiredShortPropertyName.kt
package com.nagopy.android.kotlinames.property.required

import com.nagopy.android.kotlinames.KRequiredPropertyName
import com.nagopy.android.kotlinames.KSortablePropertyName

class KRequiredShortPropertyName(val name: String) : KRequiredPropertyName<Short>, KSortablePropertyName<Short> {

    override fun name(): String = name

}
